src/main/java/com/bugsnag/Report.java

```diff
@@ -0,0 +1,202 @@
+package com.bugsnag;
+
+import com.bugsnag.serialization.Expose;
+import com.bugsnag.util.FilterTransformer;
+
+import com.google.common.collect.Maps;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class Report {
+    private static final String PAYLOAD_VERSION = "2";
+
+    private Configuration config;
+
+    private String apiKey;
+    private Throwable throwable;
+    private Severity severity = Severity.WARNING;
+    private String groupingHash;
+    private Diagnostics diagnostics = new Diagnostics();
+    private boolean shouldCancel = false;
+
+    /**
+     * Create a report for the error.
+     *
+     * @param config    the configuration for the report.
+     * @param throwable the error to create the report for.
+     */
+    public Report(Configuration config, Throwable throwable) {
```

Comment:
If the expectation is that this always get built from Bugsnag.buildReport we should remove the `public` here to make this package protected